### PR TITLE
HMRC variant of address finder

### PR DESCRIPTION
**Pull request by gavinwye**
## Background

The pull request adds the HMRC variant of the address finder to the current prototype in preparation for user research in Newcastle on the 18/4.

## Changes/additions

- Added a link to the HMRC variant to the GOV.UK address finder page - This may need to be changed once the final content is in the pattern
- removed the pattern from the main navigation by creating an 'HMRC components' section (in the front matter of the page
- added standard content to the pattern
- added a description of why this is a different pattern
- added a note to the top of the pattern explaining that it's an HMRC pattern

### CODE REVIEW

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+title:      Address lookup
+section:    HMRC Components
+status:     Approved
+discuss:    https://designpatterns.hackpad.com/CgrMSGRAhRc
+department: HMRC
+---
+
+**This is an HMRC pattern, you should only use this if the service you're designing is and HMRC service, or have the same database requirments as HMRC.**
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: "is and", "requirments"

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    </div>
+<% end %>
+
+## When to use this pattern
+
+Use an address lookup when you’re asking users for a UK address. 
+## When not to use this pattern
+
+Address lookups generally only work for UK addresses. If you’re collecting only or mostly international addresses, you should use a manual option (either multiple fields or a free text box) instead.
+How it works
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Should this be a heading?

File: source/design-patterns/patterns/components/addresses/index.html.md.erb [hunk 3]
```diff
@@ -111,3 +111,6 @@ When using an address finder you should:
 - make it clear that the address finder only works for UK addresses
 - provide a manual option for people with international addresses or addresses that are missing or badly formed in the Royal Mail database
 
+### Variants of addess finder
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/index.html.md.erb]: "addess"

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## Using an address lookup
+
+<% partial "/layouts/example" do %>
+    <div class="form-group">
+        <h1 class="heading-large">Find your address</h1>
+        <label class="form-label" for="postcode">Postcode</label>
+        <input class="form-control form-control-1-3" type="text" name="postcode" value=""><br><br>
+        <label class="form-label" for="postcode">House name or number (optional)</label>
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: `for` attribute references the previous field (postcode)

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        <input class="form-control" type="text" name="postcode" value=""><br><br>
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: `name` is `postcode` but should be e.g. `house_number`

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+**This is an HMRC pattern, you should only use this if the service you're designing is and HMRC service, or have the same database requirments as HMRC.**
+
+
+## Using an address lookup
+
+<% partial "/layouts/example" do %>
+    <div class="form-group">
+        <h1 class="heading-large">Find your address</h1>
+        <label class="form-label" for="postcode">Postcode</label>
+        <input class="form-control form-control-1-3" type="text" name="postcode" value=""><br><br>
```
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: I think this block should be e.g.

```
<h1 class="heading-large">Find your address</h1>

<div class="form-group">
  <label class="form-label" for="postcode">Postcode</label>
  <input class="form-control form-control-1-3" type="text" name="postcode" value="">
</div>

<div class="form-group">
  <label class="form-label" for="postcode">House name or number (optional)</label>
  <input class="form-control" type="text" name="postcode" value=""><br><br>
</div>
```

as in http://govuk-elements.herokuapp.com/form-elements/#form-spacing – then you don't need the `<br>`s

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb (hunk 1 above)
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Can we make sure the callout follows the same style we're currently using for Service Manual callouts (i.e. 
%This is an HMRC pattern. You should only use this if the service...%

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+section:    HMRC Components
+status:     Approved
+discuss:    https://designpatterns.hackpad.com/CgrMSGRAhRc
+department: HMRC
+---
+
+**This is an HMRC pattern, you should only use this if the service you're designing is and HMRC service, or have the same database requirments as HMRC.**
+
+
+## Using an address lookup
```
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Can we change this to ##Overview

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,47 @@
+---
+title:      Address lookup
+section:    HMRC Components
+status:     Approved
+discuss:    https://designpatterns.hackpad.com/CgrMSGRAhRc
```
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Can we insteas link to the embedded discussion at the bottom of the page with the text 'Discuss HMRC address lookup on Hackpad' - the way we have with the other patterns in the prototype?

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb (hunk 2 above)
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Change to ##How it works

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb (hunk 8 above)
Comment (36degrees) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: That's automatically added by the layout based on this config, so there's nothing to do here.

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb (hunk 1 above)
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: (and full stop after pattern.)

Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: (and change 'and HMRC service' to 'an HMRC service')

Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: (and should say 'has the same database requirements', not 'have'.)

File: source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        <input type="submit" class="button" value="Find your address">
+    </div>
+<% end %>
+
+## When to use this pattern
+
+Use an address lookup when you’re asking users for a UK address. 
+## When not to use this pattern
+
+Address lookups generally only work for UK addresses. If you’re collecting only or mostly international addresses, you should use a manual option (either multiple fields or a free text box) instead.
```
Comment (amyhupe) [inline source/design-patterns/patterns/components/addresses/hmrc_address.html.md.erb]: Can we link from 'free text box' and 'multiple fields' to the corresponding H2s in the main addresses pattern?

**Comment by 36degrees** [comment]
I'm going to merge this and make these changes on master as we need this for user research on Tuesday.

**Pull request merged**